src/org/zaproxy/zap/extension/exportReport/Export/ReportExport.java

```diff
@@ -0,0 +1,328 @@
+						if (alertDetails.get(j).equalsIgnoreCase(wascid))
+							item.setWASCID(entityEncode(Integer.toString(alert.getWascId())));
+
+						temp = alert.getMessage().getRequestHeader().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(requestheader) && temp != null	&& temp.length() > 0) {
+							item.setRequestHeader(entityEncode(temp));
+						}
+						
+						temp = alert.getMessage().getRequestBody().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(requestbody) && temp != null	&& temp.length() > 0) {
+							item.setRequestBody(entityEncode(temp));
+						}
+
+						temp = alert.getMessage().getResponseHeader().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(responseheader) && temp != null	&& temp.length() > 0) {
+							item.setResponseHeader(entityEncode(temp));
+						}
+
+						temp = alert.getMessage().getResponseBody().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(responsebody) && temp != null	&& temp.length() > 0) {
+							item.setResponseBody(entityEncode(temp));
+						}
+					}
+					
+					item.setURI(entityEncode(alert.getUri()));
+					if (alert.getParam() != null && alert.getParam().length() > 0)
+						item.setParam(entityEncode(alert.getParam()));
+					if (alert.getAttack() != null && alert.getAttack().length() > 0)
+						item.setAttack(entityEncode(alert.getAttack()));
+					if (alert.getEvidence() != null && alert.getEvidence().length() > 0)
+						item.setEvidence(entityEncode(alert.getEvidence()));
+
+					a.add(item);
+				}
+				s.setAlerts(a);
+				report.add(s);
+			}
+			javax.xml.bind.JAXBContext jc = javax.xml.bind.JAXBContext.newInstance(Report.class);
+			Marshaller jaxbMarshaller = jc.createMarshaller();
+			jaxbMarshaller.setProperty(javax.xml.bind.Marshaller.JAXB_FORMATTED_OUTPUT, true);
+			jaxbMarshaller.setProperty(javax.xml.bind.Marshaller.JAXB_ENCODING, "utf-8");
+
+			jaxbMarshaller.marshal(report, new File(path + fileName + ".dump.xml"));
+
+			return path + fileName + ".dump.xml";
+		} catch (JAXBException e) {
+			e.printStackTrace();
+		}
+		return "";
+	}
+
+	public static File TRANSFORMATION(ViewDelegate view, Logger logger, String absolutePath, String p_result, String p_source, String p_xslt) 
+	{
+		File f_result = new File(p_result);
+		StreamResult result = new StreamResult(f_result);
+
+		File f_source = new File(p_source);
+		StreamSource source = new StreamSource(f_source);
+
+		File f_xslt = new File(p_xslt);
+		StreamSource xslt = new StreamSource(f_xslt);
+
+		Transformer transformer = null;
+		try
+		{
+			transformer = TransformerFactory.newInstance().newTransformer(xslt);
+			transformer.transform(source, result);
+		}
+		catch (TransformerConfigurationException | TransformerFactoryConfigurationError e)
+		{
+			logger.error(e.getMessage(), e);
+			view.showMessageDialog(MessageFormat.format("Error: Transformer Configuration problem, see log for more details.", new Object[] { absolutePath }));
```

Comment:
It's the path that the user picked to export the file to from the JFileChooser/Writter